### PR TITLE
feat: show pending transactions

### PR DESCRIPTION
**Pull request by janek26**
this PR adds support for pending transactions and shows them. 

It also adds some other improvements:
- Indicator is shown when token balances update in the background (swr)
- when signing something the extension shows a loading screen after approving until the transaction was broadcasted

### CODE REVIEW

File: packages/extension/src/background/trackTransactions.ts [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+  TransactionStatus,
+} from "../shared/transactions.model"
+
+interface TransactionStatusWithProvider extends TransactionStatus {
+  provider: Provider
 }
+type FetchedTransactionStatus = Omit<
+  Omit<TransactionStatus, "walletAddress">,
+  "meta"
+>
```
Comment (delaaxe) [inline packages/extension/src/background/trackTransactions.ts:15]:
```suggestion
type FetchedTransactionStatus = Omit<TransactionStatus, "walletAddress" | "meta">
```

File: packages/extension/src/ui/components/Account/Transactions.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import { FC } from "react"
+import styled, { css, keyframes } from "styled-components"
+
+import { TransactionMeta as ITransactionMeta } from "../../../shared/transactions.model"
+// import { makeClickable } from "../../utils/a11y"
```
Comment (delaaxe) [inline packages/extension/src/ui/components/Account/Transactions.tsx:5]: Commented code

File: packages/extension/src/ui/components/Account/Transactions.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  cursor: auto;
+
+  &:hover,
+  &:focus {
+    background-color: #333332;
+  }
+`
+
+const PulseAnimation = keyframes`
+	0% {
```
Comment (delaaxe) [inline packages/extension/src/ui/components/Account/Transactions.tsx:37]: indentation off?

**Review by delaaxe — APPROVED** [review]

File: packages/extension/src/background/trackTransactions.ts (hunk 1 above)
Comment (janek26) [reply packages/extension/src/background/trackTransactions.ts]: thanks, didn't know that!

**Pull request merged**